data-pipeline-flink/user-cache-updater-2.0/src/main/scala/org/sunbird/dp/usercache/util/UserMetadataUpdater.scala

```diff
@@ -0,0 +1,251 @@
+    val userData: mutable.Map[String, AnyRef] = mutable.Map[String, AnyRef]()
+    Option(event.getContextDataId(cDataType = "SignupType")).map(signInType => {
+      if (config.userSelfSignedInTypeList.contains(signInType)) {
+        userData.put(config.userSignInTypeKey, config.userSelfSignedKey)
+      }
+      if (config.userValidatedTypeList.contains(signInType)) {
+        userData.put(config.userSignInTypeKey, config.userValidatedKey)
+      }
+    }).orNull
+    userData
+  }
+
+  def updateAction(userId: String, event: Event, metrics: Metrics): mutable.Map[String, AnyRef] = {
+    val userCacheData: mutable.Map[String, String] = dataCache.hgetAllWithRetry(userId)
+
+    Option(event.getContextDataId(cDataType = "UserRole")).map(loginType => {
+      userCacheData.put(config.userLoginTypeKey, loginType)
+    })
+    val userMetaDataList = event.userMetaData()
+    if (!userMetaDataList.isEmpty() && userCacheData.contains(config.userSignInTypeKey) && ("Anonymous" != userCacheData.get(config.userSignInTypeKey))) {
+
+      // Get the user details from the cassandra table
+
+      val userDetails: mutable.Map[String, AnyRef] = userCacheData.++(extractUserMetaData(readFromCassandra(
+        keyspace = config.keySpace,
+        table = config.userTable,
+        QueryBuilder.eq("id", userId),
+        metrics)
+      ))
+      val updatedMap: mutable.Map[String, AnyRef] = userDetails.+=("iscustodianuser" -> isCustodianUser(userDetails, custodianRootOrgId).asInstanceOf[AnyRef])
+      val reportInfoMap: mutable.Map[String, AnyRef] = if (updatedMap.get("iscustodianuser").get == true) getCustodianUserInfo(updatedMap, metrics, userId) else getStateUserInfo(updatedMap, metrics, userId)
+
+      logger.info(s"User details ( $userId ) are fetched from the db's and updating the redis now.")
+      reportInfoMap
+    } else {
+      logger.info(s"Skipping the event update from databases since event Does not have user properties or user sigin in type is Anonymous ")
+      userCacheData.asInstanceOf[mutable.Map[String,AnyRef]]
+    }
+  }
+
+  def getCustodianRootOrgId(): String = {
+    val custRootOrgIdQuery = QueryBuilder.select("value").from(config.keySpace, config.systemSettingsTable)
+      .where(QueryBuilder.eq("id", "custodianRootOrgId")).and(QueryBuilder.eq("field", "custodianRootOrgId")).toString
+    val custRootOrgId = cassandraConnect.findOne(custRootOrgIdQuery)
+    custRootOrgId.getString("value")
+  }
+
+  def readFromCassandra(keyspace: String, table: String, clause: Clause, metrics: Metrics): util.List[Row] = {
+    var rowSet: util.List[Row] = null
+    val query = QueryBuilder.select.all.from(keyspace, table).where(clause).toString
+    rowSet = cassandraConnect.find(query)
+    if (null != rowSet && !rowSet.isEmpty) {
+      metrics.incCounter(config.dbReadSuccessCount)
+      rowSet
+    } else {
+      metrics.incCounter(config.dbReadMissCount)
+      new util.ArrayList[Row]()
+    }
+
+  }
```

Comment:
No retry logic if a read fails?

Reply:
The retry logic is there in the cassandra util